wpengine/faustjs
# Add `faust blockset` command

## Tasks


## Description

These changes introduce a new command, `faust blockset`, for the @faustwp/cli package as well as the corresponding WP REST API endpoint to handle moving the blocks from the Next.js application into the WordPress uploads directory.

## Related Issue(s):

- https://github.com/wpengine/faustjs/issues/1522

## Testing

1. Pull down these changes: `nojira-add-blockset-command`. Run `npm i` and `npm run build` at root of monorepo.
2. Run `npm run blockset -w @faustwp/block-support-example` from the root of the monorepo - this assumes you have `plugins/faustwp/` [symlinked into a local WordPress site](https://github.com/josephfusco/symlinked-plugin-branch).
3. Observe that `wp-content/uploads/faustwp/blocks/` in your WordPress site now contains `block-a` & `block-b`.
4. Delete block-a from the Next.js application
5. Run `npm run blockset -w @faustwp/block-support-example` again
6. Observe that `wp-content/uploads/faustwp/blocks/` now only contains `block-b`.

## Demo

https://github.com/wpengine/faustjs/assets/6676674/27dc59c3-eb83-4c89-8ed7-c4ca8dbc4168
